bears/general/KeywordBear.py

```diff
@@ -2,7 +2,50 @@
 
 from coalib.bearlib import deprecate_settings
 from coalib.bears.LocalBear import LocalBear
+from coalib.results.Diff import Diff
 from coalib.results.Result import RESULT_SEVERITY, Result
+from coalib.results.SourceRange import SourceRange
+
+from bears.general.AnnotationBear import AnnotationBear
+
+
+def _get_comments(dependency_results):
+    if not dependency_results:
+        return
+
+    annotation_bear_results = dependency_results.get('AnnotationBear')
+    if (not annotation_bear_results or
+            not isinstance(annotation_bear_results, list)):
+        return
+
+    for result in annotation_bear_results:
+        yield from result.contents.get('comments', [])
+
+
+def generate_diff(comments, file, filename,
+                  line, line_number, pos):
+    diffs = {}
+    todo_source_range = SourceRange.from_values(filename, line_number + 1,
+                                                pos + 1)
+    for comment_sourcerange in comments:
+        if todo_source_range not in comment_sourcerange:
+            continue
+
+        diff = Diff(file)
+        comment_start = comment_sourcerange.start.column
+        line_before_todo_comment = line[:comment_start - 1].strip()
```

Comment:
this code fails in those cases:

```C
int a = 0; /* TODO test
another test
*/
```

will render the comment invalid.

Also in this case:

```C
int a = 1; /* TODO test */ int b=2;
```

it will remove source code. This should be fixable.